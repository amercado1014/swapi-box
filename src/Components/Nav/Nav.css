.nav {
  display: flex;
  justify-content: center;
}

button {
  background-color: #feda4a;
  height: 45px;
  width: 120px;
  border: none;
  font-family: 'Anton', sans-serif;
  font-size: 20px;
  margin-left: 20px;
  cursor: pointer;
}

button:focus {
  outline: 0;
}

button:hover {
  background-color: lightgrey;
}

.people-button {
  border: 4px solid #00ff7f;
}

.people-button:focus {
  background-color: #00ff7f;
}

.planets-button {
  border: 4px solid #87ceeb;
}

.planets-button:focus {
  background-color: #87ceeb;
}

.vehicles-button {
  border: 4px solid #e35757;
}

.vehicles-button:focus {
  background-color: #e35757;
}

.favorites-button {
  border: 4px solid #a5a6ad;
  width: 145px;
}

.favorites-button:focus {
  background-color:  dimgray;
}


