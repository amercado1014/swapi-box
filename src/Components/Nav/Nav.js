import React from 'react';
import './Nav.css';

const Nav = ({fetchCards, favorites, displayFavorites}) => {
  return (
    <div>
      <button 
        onClick={() => fetchCards('people', 'peopleData')}
      >People</button>
      <button 
        onClick={() => fetchCards('planets', 'planetData')}
      >Planets</button>
      <button 
        onClick={() => fetchCards('vehicles', 'vehicleData')}
      >Vehicles</button>
      <button
        onClick={() => displayFavorites('favorites')}
      >Favorites: <span>{favorites.length}</span></button>
    </div>
  );
};

export default Nav;