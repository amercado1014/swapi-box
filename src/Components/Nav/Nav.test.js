import React from 'react';
import Nav from './Nav';
import { shallow } from 'enzyme';

describe('Nav', () => {
  let wrapper;
  let mockFetchCards;
  let mockDisplayFavorites;
  let mockFavorites;


  beforeEach(() => {
    mockFetchCards = jest.fn();
    mockDisplayFavorites = jest.fn();
    mockFavorites = [
      {
        name: "Luke Skywalker",
        class: "people-card",
        data: {
          homeworld: "Tatooine",
          language: "Galactic Basic",
          population: "200000",
          species: "Human"
        }
      }
    ];
    wrapper = shallow(<Nav fetchCards={mockFetchCards}
      favorites={mockFavorites}
      displayFavorites={mockDisplayFavorites} />);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should display the length of favorites array', () => {
    expect(wrapper.find('span').text()).toEqual('1');
  });

  it('should invoke fetchCards with the right params', () => {
    const peopleButton = wrapper.find('.people-button');
    peopleButton.simulate('click');
    expect(mockFetchCards).toHaveBeenCalledWith('people', 'peopleData');
  });

  it('should invoke displayFavorites with the right params', () => {
    const favoritesButton = wrapper.find('.favorites-button');
    favoritesButton.simulate('click');
    expect(mockDisplayFavorites).toHaveBeenCalledWith('favorites');
  });
});