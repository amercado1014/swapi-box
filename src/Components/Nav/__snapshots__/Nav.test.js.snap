// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    displayFavorites={[Function]}
    favorites={
        Array [
            Object {
              "class": "people-card",
              "data": Object {
                "homeworld": "Tatooine",
                "language": "Galactic Basic",
                "population": "200000",
                "species": "Human",
              },
              "name": "Luke Skywalker",
            },
          ]
    }
    fetchCards={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          onClick={[Function]}
>
          People
</button>,
        <button
          onClick={[Function]}
>
          Planets
</button>,
        <button
          onClick={[Function]}
>
          Vehicles
</button>,
        <button
          onClick={[Function]}
>
          Favorites: 
          <span>
                    1
          </span>
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "People",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "People",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planets",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicles",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Favorites: ",
            <span>
              1
</span>,
          ],
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Favorites: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 1,
            },
            "ref": null,
            "rendered": 1,
            "type": "span",
          },
        ],
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            onClick={[Function]}
>
            People
</button>,
          <button
            onClick={[Function]}
>
            Planets
</button>,
          <button
            onClick={[Function]}
>
            Vehicles
</button>,
          <button
            onClick={[Function]}
>
            Favorites: 
            <span>
                        1
            </span>
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Favorites: ",
              <span>
                1
</span>,
            ],
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Favorites: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": "span",
            },
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
